### PR TITLE
Fix #6426: completion doesn't distinguish between implied and normal imports

### LINKED ISSUE #6426: REPL code completion for imports does not take into account the `implied` modifier
```scala
scala> object Foo {
     |   val explicitVal: Int = 1
     |   implied implicitVal for Int = 2
     | }
scala> import Foo.implicit<TAB>  // Completes to `implicitVal`
scala> import implied Foo.explicit<TAB> // Completes to `explicitVal`
```

See [this](https://github.com/lampepfl/dotty/blob/187f6a27b55cfc1aa57ada1df9d5de2a1d2a8fe2/compiler/src/dotty/tools/dotc/interactive/Completion.scala#L124) TODO comment.

### PR DESCRIPTION
**Pull request by anatoliykmetyuk**

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/interactive/Completion.scala [hunk 1]
```diff
@@ -33,6 +33,12 @@ import scala.collection.mutable
  */
 case class Completion(label: String, description: String, symbols: List[Symbol])
 
+/** Mode of code completion specifying which members it can see: all, explicit or implied */
+enum ImportMode { case All, Explicit, Implied }
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/interactive/Completion.scala]: Can't use a dotty enum in the compiler until we drop the 2.12 CI: https://github.com/lampepfl/dotty/blob/9a2d2d1f0dac8618832dda202cf1946c30991001/.drone.yml#L60 :)

Comment (anatoliykmetyuk) [reply compiler/src/dotty/tools/dotc/interactive/Completion.scala]: Is there any issue blocking that? Or can I just drop it?

Comment (smarter) [reply compiler/src/dotty/tools/dotc/interactive/Completion.scala]: We wanted to wait a bit and see if any blocker forced us to revert to 2.12. Let's discuss on Monday if it's time to drop 2.12 completely.

Comment (anatoliykmetyuk) [reply compiler/src/dotty/tools/dotc/interactive/Completion.scala]: Yes, this new commit is not pretty

**Pull request closed without merging**